src/hotspot/share/opto/memnode.cpp

```diff
@@ -926,6 +926,7 @@ Node* LoadNode::make(PhaseGVN& gvn, Node* ctl, Node* mem, Node* adr, const TypeP
   case T_DOUBLE:  load = new LoadDNode (ctl, mem, adr, adr_type, rt,            mo, control_dependency, require_atomic_access); break;
   case T_ADDRESS: load = new LoadPNode (ctl, mem, adr, adr_type, rt->is_ptr(),  mo, control_dependency); break;
   case T_OBJECT:
+  case T_NARROWOOP:
```

Comment:
Why is this addition needed for your PR?

Reply:
Apparently C2 will work if you use "T_OBJECT" in places when the type is actually "T_NARROWOOP", see [this comment here](https://github.com/openjdk/jdk/blob/065d3e0d58c96b8a84f3c02bb8704fab6459eaa7/src/hotspot/share/opto/graphKit.cpp#L1772). However, I preferred adding a new type to the switch table.